db/seeds.rb

```diff
@@ -5,3 +5,468 @@
+
+def task_gen(location)
+  task = Task.new(location: location)
+  task.name = short_phrase_gen
+  task.kind = ["Hourly", "Daily", "Weekly"].sample
+  task.start = Date.today
+  task.end = Date.today+Total_days.days
+  task.time_of_day = Faker::Time.between(Time.now.beginning_of_day, Time.now.end_of_day, :all)
+  task.day_in_week = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"].sample
+  task.active = true
+  task.description = Faker::Lorem.sentence
+  task.link = Faker::Internet.url
+  task.save!
+end
+
+def data_fields_gen(type)
+  type.data_fields.create!(name: "How many boxes of paper?", display_type: "text_field", values: "integer", upper_bound: 10.0, lower_bound: 1.0, exact_alert: "0", active: true)
+  type.data_fields.create!(name: "Work Notes", display_type: "text_area", active: true)
+  type.data_fields.create!(name: "Which college?", display_type: "select", values: "BR, JE, SY, TC, TD, SM, MC, ES, PC, DC, BC, CC", exact_alert: "JE", active: true)
+  type.data_fields.create!(name: "Mac or PC?", display_type: "radio_button", values: "Mac, PC", exact_alert: "PC", active: true)
+  type.data_fields.create!(name: "Are you happy?", display_type: "check_box", values: "yes, no, dunno", exact_alert: "no", active: true)
+end
+
+def data_type_gen
+  type = DataType.new(department: @department)
+  type.name = Faker::Hacker.noun
+  type.description = Faker::Lorem.sentence
+  type.save!
+  data_fields_gen(type)
+  type
+end
+
+def data_object_gen(type)
+  object = DataObject.new(data_type: type, name: short_phrase_gen, description: Faker::Hacker.adjective)
+  object.locations = Location.all
+  object.save!
+end
+
+def category_gen
+  cat = Category.new(name: short_phrase_gen, department: @department, billing_code: ptaeo_gen)
+  cat.save!
+end
+
+def payform_item_gen(payform, cat)
+  item = PayformItem.new(category: cat, payform: payform, date: Date.today, source: payform.user.name)
+  item.hours = 1 + rand(Payform_item_max_hour)
+  item.description = Faker::Hacker.say_something_smart
+  item.save!
+end
+
+def print_payforms    
+  set = PayformSet.new(department: @department)
+  set.payforms = @department.payforms.unprinted
+  set.payforms.map {|p| p.printed = Time.now}
+  set.save!
+end
+
+def report_item_gen(report, ip)
+  report.report_items.create!(time: Time.now, content: Faker::Hacker.say_something_smart, ip_address: ip)
+end
+
+def report_gen(shift)
+  report = Report.new(shift: shift, arrived: shift.start)
+  report.save!
+  ip = Faker::Internet.ip_v4_address
+  NReport_item_per_report.times do 
+    report_item_gen(report, ip)
+  end
+  report.update_attributes(departed: shift.end)
+end
+
+################################################################################
+## Beginning of seed script
+################################################################################
+Timecop.travel(DateTime.now - Ndays_of_schedule_into_past.days)
+
+# First AppConfig
+progress_bar_gen("AppConfig [1/15]", 1) do
+  @app_config = app_config_gen
```

Comment:
To make it accessible in the entire script, including when inside methods. As this is the practice both within the app (controller/view) and within the test (helper/spec) for these ActiveRecord variables, I have adopted the same thing here in the seed script